src/tui/editor.zig

```diff
@@ -2109,6 +2109,16 @@ pub const Editor = struct {
     }
     pub const paste_meta = .{ .description = "Paste from internal clipboard" };
 
+    pub fn delete_line(self: *Self, ctx: Context) Result {
+        try self.move_begin(ctx);
+        const b = try self.buf_for_update();
+        const root = try self.delete_to(move_cursor_end, b.root, b.allocator);
+        try self.update_buf(root);
+        self.clamp();
+        try self.delete_backward(ctx);
```

Comment:
I think this should probably be `delete_forwards` to leave the cursor at the beginning of the *next* line like in vim. Also, you probably want to integrate the `delete_forward` actions here before the `update_buf` call instead of just calling it, otherwise you'll create two undo steps instead of one.

Reply:
Btw, both vim and nvim yank to the clipboard on `dd`, which is really useful.

The existing `cut` command in in flow is exactly the same as `dd` in vim/nvim when there is no selection, and `d` when there is one. It cuts the whole line to the clipboard if nothing is selected and cuts just the selection if there is one.

You could probably just use `cut` as-is.

Reply:
Didn't know about cut, I'll use that for `dd` thanks. I might still leave a `delete_line` command for use cases where you want to delete a line and not copy it (ie, `"_dd` in vim)